Patch that causes "set breakpoint pending" to be respected in MI mode. This
does not effect the "-break-insert -f" option.

Signed-off-by: Chris Smith<chris.smith@st.com>
Index: gdb-6.8/gdb/breakpoint.c
===================================================================
--- gdb-6.8.orig/gdb/breakpoint.c	2008-09-11 16:26:40.000000000 +0100
+++ gdb-6.8/gdb/breakpoint.c	2008-09-11 16:31:12.000000000 +0100
@@ -223,11 +223,6 @@
 		    value);
 }
 
-/* If AUTO_BOOLEAN_FALSE, gdb will not attempt to create pending breakpoints.
-   If AUTO_BOOLEAN_TRUE, gdb will automatically create pending breakpoints
-   for unrecognized breakpoint locations.  
-   If AUTO_BOOLEAN_AUTO, gdb will query when breakpoints are unrecognized.  */
-static enum auto_boolean pending_break_support;
 static void
 show_pending_break_support (struct ui_file *file, int from_tty,
 			    struct cmd_list_element *c,
Index: gdb-6.8/gdb/breakpoint.h
===================================================================
--- gdb-6.8.orig/gdb/breakpoint.h	2008-09-11 16:30:00.000000000 +0100
+++ gdb-6.8/gdb/breakpoint.h	2008-09-11 16:31:12.000000000 +0100
@@ -171,6 +171,11 @@
     hw_execute = 3		/* Execute HW breakpoint */
   };
 
+/* If AUTO_BOOLEAN_FALSE, gdb will not attempt to create pending breakpoints.
+   If AUTO_BOOLEAN_TRUE, gdb will automatically create pending breakpoints
+   for unrecognized breakpoint locations.
+   If AUTO_BOOLEAN_AUTO, gdb will query when breakpoints are unrecognized.  */
+enum auto_boolean pending_break_support;
 
 /* Information used by targets to insert and remove breakpoints.  */
 
Index: gdb-6.8/gdb/mi/mi-cmd-break.c
===================================================================
--- gdb-6.8.orig/gdb/mi/mi-cmd-break.c	2008-09-11 16:26:40.000000000 +0100
+++ gdb-6.8/gdb/mi/mi-cmd-break.c	2008-09-11 16:31:12.000000000 +0100
@@ -125,6 +125,10 @@
 	}
     }
 
+  /* Respect "set breakpoint pending" */
+  if (pending_break_support != AUTO_BOOLEAN_FALSE)
+    pending = 1;
+
   if (optind >= argc)
     error (_("mi_cmd_break_insert: Missing <location>"));
   if (optind < argc - 1)
